tests/utils/test_yaml.py

```diff
@@ -84,3 +85,22 @@ def test_read_yaml_file_bad_decode(tmpdir, caplog):
     with pytest.raises(ValueError):
         read_yaml_from_file_path(config_path, 'schemas/container.json')
     assert "unable to decode JSON schema, cannot validate" in caplog.text
+
+
+@pytest.mark.parametrize('config', [
+    ("""\
+        operator_manifests:
+            enable_digest_pinning: true
+            repo_replacements: []
+    """),
+    ("""\
+        compose:
+            packages: []
+            pulp_repos: true
+        mage_build_method: "imagebuilder"
+    """),
+])
+def test_read_yaml_validation_error(config, caplog):
+    with pytest.raises(OsbsValidationException):
+        read_yaml(config, 'schemas/container.json')
+    assert "at top level: validating 'anyOf' has failed" in caplog.text
```

Comment:
I'd like to assert whole message and from exception not from log. I care more about that exception than logs.